pkg/resource/provider/component_provider.go

```diff
@@ -32,6 +32,33 @@ type componentProvider struct {
 	version   string
 	schema    []byte
 	construct provider.ConstructFunc
+	call      provider.CallFunc
+}
+
+type Options struct {
+	Name      string
+	Version   string
+	Schema    []byte
+	Construct provider.ConstructFunc
+	Call      provider.CallFunc
+}
+
+// ProviderMain is an entrypoint for a resource provider plugin that implements `Construct` and optionally also
+// `Call` for component resources.
+//
+// Using it isn't required but can cut down significantly on the amount of boilerplate necessary to fire up a new
+// resource provider for components.
+func ProviderMain(opts Options) error {
```

Comment:
Looks like there's a lint warning:

> resource/provider/component_provider.go:51:6: func name will be used as provider.ProviderMain by other packages, and that stutters; consider calling this Main (golint)

We already have `Main`, so we could ignore the lint warning.

Another option is to rename this to something like `MainWithOptions` (is that the typical naming convention when adding such an "overload" in Go?).

Reply:
I don't really have any opinion on which way to go here.

Reply:
`MainWithOptions` seems good to me!

Reply:
When you get a chance - can get get this last comment addressed @jen20? Would love to get this PR merged in, thanks!

Reply:
I've adjusted the code and commit message, and force pushed a new commit.